tests/test_utils/test_schemaview.py

```diff
@@ -2281,3 +2283,144 @@ def test_class_name_mappings() -> None:
     assert set(view.all_slots()) == set(slot_names)
     assert set(view.slot_name_mappings()) == set(slot_names.values())
     assert {snm_def.name: snm for snm, snm_def in view.slot_name_mappings().items()} == slot_names
+
+
+@pytest.fixture(scope="module")
+def sv_cycles_schema() -> SchemaView:
+    """A schema containing cycles!"""
+    return SchemaView(INPUT_DIR_PATH / "cycles.yaml")
+
+
+CYCLES = {
+    TYPES: {
+        # types in cycles, either directly or via ancestors
+        # key: type, value: node where the cycle starts
+        0: {
+            "curve_type": "circular_type",
+            "semi_circular_type": "circular_type",
+            "circular_type": "circular_type",
+            "type_circular": "type_circular",
+            "circle": "circle",
+            "circle_of_life": "circle",
+        },
+        # types not involved in cycles
+        # key: the type, value: ancestors of the type
+        1: {
+            "supreme_string": {"supreme_string", "super_string", "string"},
+            "super_string": {"super_string", "string"},
+            "string": {"string"},
+            "integer": {"integer"},
+            "boolean": {"boolean"},
+        },
+    },
+    CLASSES: {
+        # classes involved in cycles
+        # key: class name, value: node where the cycle starts
+        0: {
+            "ClassA": "ClassA",
+            "ClassB": "ClassB",
+            "ClassC": "ClassC",
+            "ClassD": "ClassA",
+            "ClassE": "ClassA",
+            "ClassF": "ClassF",
+            "ClassG": "ClassF",
+            "Mixin1": "Mixin1",
+            "Mixin2": "Mixin2",
+            "MixedClass": "Mixin2",
+        },
+        # class ancestors for classes not in cycles
+        # key: class name, value: class ancestors
+        1: {
+            "BaseClass": {"BaseClass"},
+            "MixinA": {"MixinA"},
+            "MixinB": {"MixinB"},
+            "NonCycleClassA": {"NonCycleClassA", "BaseClass"},
+            "NonCycleClassB": {"MixinA", "NonCycleClassB", "NonCycleClassA", "BaseClass"},
+            "NonCycleClassC": {"MixinB", "NonCycleClassC", "NonCycleClassA", "BaseClass"},
+            "IdentifierCycleClassA": {"IdentifierCycleClassA"},
+            "IdentifierCycleClassB": {"IdentifierCycleClassB"},
+            "IdentifierCycleClassC": {"IdentifierCycleClassC"},
+            "IdentifierCycleClassD": {"IdentifierCycleClassD"},
+        },
+    },
+}
+
+
+@pytest.mark.parametrize(("target", "cycle_start_node"), [(k, v) for k, v in CYCLES[TYPES][0].items()])
+@pytest.mark.parametrize("fn", ["detect_cycles", "graph_closure", "type_ancestors"])
```

Comment:
test calling the function directly, via the graph closure (`_closure` function), and through SchemaView methods that use `_closure`